HiringApp/HiringAppCore/Repositories/Repository.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  TechnologiesRepository.swift
+//  Pods
+//
+//  Created by Alba Luján on 21/6/17.
+//
+//
+
+import Foundation
+import Deferred
+import BSWFoundation
+
+enum RepositoryError: Error {
+    case noData
+    case cantSave
+}
+
+class Repository {
+    
+    //MARK: - Stored properties
+    let apiProvider: APIProviderType
+    let cacheProvider: CacheProviderType
+    let dbProvider: DBProviderType
+    
+    init(apiProvider: APIProviderType, cacheProvider: CacheProviderType, dbProvider: DBProviderType) {
+        self.apiProvider = apiProvider
+        self.cacheProvider = cacheProvider
+        self.dbProvider = dbProvider
+    }
+    
+    func retrieveTechnologies() -> Task<[TechnologyModel]> {
+        if let technologies = cacheProvider.getTechnologies() {
+            return Task(success: technologies)
+        }
+                
+        let techTask = apiProvider.retrieveTechnologies().andThen(upon: .main) { (modelsArray) -> Task<[TechnologyModel]> in
+            return self.cacheProvider
+                .saveTechnologies(technologies: modelsArray) //Save modelsArray into cache
+                .recover(upon: .main, substituting: { (_) in })
+                .map(upon: .main, transform: { (_) -> [TechnologyModel] in
+                    return modelsArray
+                })
+        }
+        
+        return techTask
+    }
+    
+    func retrieveDBTechnologies() -> Task<[TechnologyModel]> {
```

Comment:
this naming is a bit weird... considering there's also another function called the same... what does it do differently? are both useful?